nixos/tests/sourcehut/git.nix

```diff
@@ -0,0 +1,98 @@
+import ../make-test-python.nix ({ pkgs, lib, ... }:
+let
+  domain = "sourcehut.localdomain";
+in
+{
+  name = "sourcehut";
+
+  meta.maintainers = with pkgs.lib.maintainers; [ tomberek nessdoor ];
+
+  nodes.machine = { config, pkgs, nodes, ... }: {
+    imports = [
+      ./nodes/common.nix
+    ];
+
+    networking.domain = domain;
+    networking.extraHosts = ''
+      ${config.networking.primaryIPAddress} git.${domain}
+      ${config.networking.primaryIPAddress} meta.${domain}
+    '';
+
+    services.sourcehut = {
+      git.enable = true;
+      settings."git.sr.ht" = {
+        oauth-client-secret = pkgs.writeText "gitsrht-oauth-client-secret" "3597288dc2c716e567db5384f493b09d";
+        oauth-client-id = "d07cb713d920702e";
+      };
+    };
+
+    environment.systemPackages = with pkgs; [
+      git
+    ];
+  };
+
+  testScript =
+    let
+      userName = "nixos-test";
+      userPass = "AutoNixosTestPwd";
+      hutConfig = pkgs.writeText "hut-config" ''
+        instance "${domain}" {
+          # Will be replaced at runtime with the generated token
+          access-token "OAUTH-TOKEN"
+        }
+      '';
+      sshConfig = pkgs.writeText "ssh-config" ''
+        Host git.${domain}
+             IdentityFile = ~/.ssh/id_rsa
+      '';
+    in
+    ''
+      start_all()
+      machine.wait_for_unit("multi-user.target")
+      # For some reason, OpenSSH fails to start on boot with code 226/NAMESPACE
+      machine.systemctl("start sshd.service")
```

Comment:
```suggestion
```

This actually broke the testsuite completely for me, as the failed ssh service caused the test VM to abort. Also, this shouldn't make it into the testsuite anyway, IMO.

It seems to happen if `/var/lib/sourcehut/gitsrht/repos` (or whatever path the user configured through `services.sourcehut.settings."git.sr.ht".repos`) does not exist when ssh is started, thus it cannot be mounted into the service namespace.

This could (or rather: should?) be solved by ensuring the directory exists, by using `systemd.tmpfiles`. I did the changes locally and it fixes the issue, at least I could not reproduce it anymore. The full diff of my changes:

```diff
diff --git a/nixos/modules/services/misc/sourcehut/default.nix b/nixos/modules/services/misc/sourcehut/default.nix
index aa803d3bb693..f002635d0bb3 100644
--- a/nixos/modules/services/misc/sourcehut/default.nix
+++ b/nixos/modules/services/misc/sourcehut/default.nix
@@ -790,13 +790,21 @@ in
         '';
       };
       systemd.tmpfiles.settings."10-sourcehut-gitsrht" = mkIf cfg.git.enable (
-        builtins.listToAttrs (map (name: {
-          name = "/var/log/sourcehut/gitsrht-${name}";
-          value.f = {
-            inherit (cfg.git) user group;
-            mode = "0644";
-          };
-        }) [ "keys" "shell" "update-hook" ])
+        mkMerge [
+          (builtins.listToAttrs (map (name: {
+            name = "/var/log/sourcehut/gitsrht-${name}";
+            value.f = {
+              inherit (cfg.git) user group;
+              mode = "0644";
+            };
+          }) [ "keys" "shell" "update-hook" ]))
+          {
+            ${cfg.settings."git.sr.ht".repos}.d = {
+              inherit (cfg.git) user group;
+              mode = "0644";
+            };
+          }
+        ]
       );
       systemd.services.sshd = {
         preStart = mkIf cfg.hg.enable ''
diff --git a/nixos/tests/sourcehut/git.nix b/nixos/tests/sourcehut/git.nix
index cb3ca57ddf62..ed184d5d5518 100644
--- a/nixos/tests/sourcehut/git.nix
+++ b/nixos/tests/sourcehut/git.nix
@@ -49,8 +49,6 @@ in
     ''
       start_all()
       machine.wait_for_unit("multi-user.target")
-      # For some reason, OpenSSH fails to start on boot with code 226/NAMESPACE
-      machine.systemctl("start sshd.service")
       machine.wait_for_unit("sshd.service")

       with subtest("Check whether meta comes up"):
```

Reply:
> Should we integrate the fix as part of this PR or as a separate one?

To be honest, I'd just integrate it as part of this PR. It uncovered this bug and more-or-less depends on the fix to properly work, so I don't see why not. :+1: 

> That was my original intention, but since you already did the heavylifting...

I just investigated it, since it broke the testsuite on my machine as a whole, i.e. if would not even get as far as the above workaround. Didn't want to step on anyones toe, though.

Reply:
Oh no no no, you didn't step on anybody's toes, I'm in fact happy to not be the only one working on this thing! I'll integrate your changes immediately, then.